src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserController.kt

```diff
@@ -84,6 +85,45 @@ class UserController(private val userSearchService: UserSearchService, private v
     return if (users.count() == 0) ResponseEntity.noContent().build() else ResponseEntity.ok(users)
   }
 
+  @GetMapping("/{username}")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "User detail.",
+    description = "User detail."
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "User not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  suspend fun user(
+    @Parameter(description = "The username of the user.", required = true) @PathVariable
+    username: String
+  ) = UserDto.fromUser(userService.getAuthUserByUsername(username) ?: throw UsernameNotFoundException("Account for username $username not found"))
```

Comment:
Cleaner to move the null check and exception throw down into the service?

Reply:
Moved to Service